bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BatchExecutor.java

```diff
@@ -155,33 +156,47 @@ public BatchExecutor(
   private <R extends Row, T> ListenableFuture<Result> issueAsyncRowRequest(Row row,
       Batch.Callback<T> callback, Object[] results, int index) throws InterruptedException {
     LOG.trace("issueRowRequest(Row, Batch.Callback, Object[], index");
-    ListenableFuture<? extends GeneratedMessage> future = issueAsyncRequest(row);
-
     SettableFuture<Result> resultFuture = SettableFuture.create();
-    results[index] = null;
     RpcResultFutureCallback<T> futureCallback =
         new RpcResultFutureCallback<T>(row, callback, index, results, resultFuture);
-    Futures.addCallback(future, futureCallback, service);
+    results[index] = null;
+    if (service.isShutdown()) {
```

Comment:
This looks good.
Is there a unit test where we test this behavior?

if (closed)
  expect ConnectionClosedException
else
  expect async all